### PR TITLE
Add 3D checkpoint visualization

### PR DESCRIPTION
**Pull request by riidefi**
Footage: https://youtu.be/NSwiO4purMA

### CODE REVIEW

File: assets/message/MenuSP_U.bmg.json5 [hunk 1]
```diff
@@ -1095,6 +1095,22 @@
         "font": "regular",
         "string": "No track was found in the\n/mkw-sp/thumbnails folder.",
     },
+    "10315": {
+        "font": "regular",
+        "string": "Debug overlay",
```
Comment (vabold) [inline assets/message/MenuSP_U.bmg.json5:1100]: Capitalize "Overlay".

Comment (riidefi) [reply assets/message/MenuSP_U.bmg.json5]: Fixed

File: payload/sp/settings/ClientSettings.cc [hunk 2]
```diff
@@ -6,7 +6,10 @@ namespace SP::ClientSettings {
 
 const char name[] = "MKW-SP Settings";
 
-const u32 categoryMessageIds[] = { 10118, 10126, 10119, 10262, 10258, 10120 };
+const u32 categoryMessageIds[] = { 10118, 10126, 10119, 10262, 10258, 10120, 10315 };
+
+constexpr u32 MID_DISABLE = 10057;
+constexpr u32 MID_ENABLE = 10058;
```
Comment (stblr) [inline payload/sp/settings/ClientSettings.cc:12]: We actually need one message per setting because of i18n.

File: payload/sp/3d/Checkpoints.cc [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+                0x00, 0x10, 0x08, 0x00, 0x00, 0x00, 0x01, 0x08, 0x70, 0x40, 0x77, 0x70, 0x09, 0x08,
+                0x80, 0xC8, 0x24, 0x12, 0x09, 0x08, 0x90, 0x04, 0x82, 0x41, 0x20, 0x00, 0x10, 0x00,
+                0x00, 0x10, 0x09, 0x00, 0x00, 0x00, 0x01, 0x10, 0x00, 0x00, 0x10, 0x0E, 0x00, 0x00,
+                0x05, 0x01, 0x10, 0x00, 0x00, 0x10, 0x10, 0x00, 0x00, 0x05, 0x01, 0x10, 0x00, 0x00,
+                0x10, 0x3F, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+                0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00};
+
+        mDl = &CACHED_DL;
+        mDlSize = 0xA0;
+    }
```
Comment (stblr) [inline payload/sp/3d/Checkpoints.cc:71]: Any reason to have this rather than a static constant?

File: payload/sp/3d/Checkpoints.cc [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+                0x08, 0x50, 0x00, 0x00, 0x22, 0x00, 0x08, 0x60, 0x00, 0x00, 0x00, 0x00, 0x10, 0x00,
+                0x00, 0x10, 0x08, 0x00, 0x00, 0x00, 0x01, 0x08, 0x70, 0x40, 0x77, 0x70, 0x09, 0x08,
+                0x80, 0xC8, 0x24, 0x12, 0x09, 0x08, 0x90, 0x04, 0x82, 0x41, 0x20, 0x00, 0x10, 0x00,
+                0x00, 0x10, 0x09, 0x00, 0x00, 0x00, 0x01, 0x10, 0x00, 0x00, 0x10, 0x0E, 0x00, 0x00,
+                0x05, 0x01, 0x10, 0x00, 0x00, 0x10, 0x10, 0x00, 0x00, 0x05, 0x01, 0x10, 0x00, 0x00,
+                0x10, 0x3F, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+                0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00};
+
+        mDl = &CACHED_DL;
+        mDlSize = 0xA0;
```
Comment (stblr) [inline payload/sp/3d/Checkpoints.cc:70]: `std::size` or `std::array`

File: payload/sp/3d/Checkpoints.cc [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+    return System::CourseMap::spInstance->mpCheckPath;
+}
+
+struct TranslucentVertexColors {
+    TranslucentVertexColors() {
+        createDisplayListCached();
+    }
+
+    void createDisplayListCached() {
+        alignas(32) static u8 CACHED_DL[] = {0x61, 0x28, 0x04, 0x90, 0x00, 0x61, 0xC0, 0x08, 0xA8,
```
Comment (stblr) [inline payload/sp/3d/Checkpoints.cc:56]: If you have command annotations on hand, I think they should be added. Otherwise don't bother.

**Review by stblr — CHANGES_REQUESTED** [review]

File: payload/sp/settings/ClientSettings.cc (hunk 2 above)
Comment (riidefi) [reply payload/sp/settings/ClientSettings.cc:12]: Do the three existing uses of each need to be changed too?

Comment (stblr) [reply payload/sp/settings/ClientSettings.cc:12]: Yes (not necessarily in this PR).

File: payload/sp/3d/Checkpoints.cc (hunk 3 above)
Comment (riidefi) [reply payload/sp/3d/Checkpoints.cc:71]: This would allow us to quickly swap the material via a setting in the future.

**Review by stblr — APPROVED** [review]

**Pull request merged**